### PR TITLE
pull: fix a couple of issues

### PR DESCRIPTION
**Pull request by giuseppe**
Closes: https://github.com/containers/libpod/issues/2785

Closes: https://github.com/containers/libpod/issues/2710

### CODE REVIEW

File: cmd/podman/pull.go [hunk 1]
```diff
@@ -163,7 +168,7 @@ func pullCmd(c *cliconfig.PullValues) error {
 		for _, name := range names {
 			newImage, err := runtime.New(getContext(), name, c.String("signature-policy"), authfile, writer, &dockerRegistryOptions, image.SigningOptions{}, true, nil)
 			if err != nil {
-				println(errors.Wrapf(err, "error pulling image %q", name))
+				logrus.Errorf("error pulling image %q", name)
```
Comment (mheon) [inline cmd/podman/pull.go]: We still get proper verbose errors with log-level=debug enabled, right?

**Comment by mheon** [comment]
Test failures and one question, otherwise LGTM

File: cmd/podman/pull.go (hunk 1 above)
Comment (giuseppe) [reply cmd/podman/pull.go]: yes, we get the same errors as before.  The println(...) statement was only causing to print something like "(0x1840f00,0xc00041bba0)"

File: cmd/podman/pull.go [hunk 2]
```diff
@@ -61,7 +61,12 @@ func init() {
 
 // pullCmd gets the data from the command line and calls pullImage
 // to copy an image from a registry to a local machine
-func pullCmd(c *cliconfig.PullValues) error {
+func pullCmd(c *cliconfig.PullValues) (retError error) {
+	defer func() {
+		if retError != nil && exitCode == 0 {
+			exitCode = 1
```
Comment (rhatdan) [inline cmd/podman/pull.go]: ExitCode should be 125

Comment (rhatdan) [reply cmd/podman/pull.go]: We should make this a constant and start using it.

PODMANFAILED or something.

Comment (giuseppe) [reply cmd/podman/pull.go]: I get 125:

```
$ bin/podman pull fooo/bar ; echo $?
Trying to pull docker://fooo/bar...ERRO[0001] Error pulling image ref //fooo/bar:latest: Error determining manifest MIME type for docker://fooo/bar:latest: Error reading manifest latest in docker.io/fooo/bar: errors:
denied: requested access to the resource is denied
unauthorized: authentication required
 
Failed
ERRO[0001] error pulling image "docker://fooo/bar"      
Error: error pulling image "fooo/bar": Invalid image name "fooo/bar", expected colon-separated transport:reference
125

```

**Pull request merged**